pkg/core/fs_ro_ops.go

```diff
@@ -24,35 +24,130 @@ func (fs *readOnlyFsInternal) StatFS(
 	return statFS()
 }
 
-func (fs *readOnlyFsInternal) LookUpInode(ctx context.Context, op *fuseops.LookUpInodeOp) error {
-	log.Print(fmt.Printf("lookup parent id:%d, child: %s ", op.Parent, op.Name))
+func (fs *readOnlyFsInternal) opStart(op interface{}) {
+	opType := reflect.TypeOf(op)
+	switch op.(type) {
+	case *fuseops.ReadFileOp:
+		r := op.(*fuseops.ReadFileOp)
+		fs.l.Info("Start",
+			zap.String("Request", opType.String()),
+			zap.String("repo", fs.bundle.RepoID),
+			zap.String("bundle", fs.bundle.BundleID),
+			zap.Uint64("inode", uint64(r.Inode)),
+		)
+		return
+	case *fuseops.WriteFileOp:
+		w := op.(*fuseops.WriteFileOp)
+		fs.l.Info("Start",
+			zap.String("Request", opType.String()),
+			zap.String("repo", fs.bundle.RepoID),
+			zap.String("bundle", fs.bundle.BundleID),
+			zap.Uint64("inode", uint64(w.Inode)),
+		)
+		return
+	case *fuseops.ReadDirOp:
+		r := op.(*fuseops.ReadDirOp)
+		fs.l.Info("Start",
+			zap.String("Request", opType.String()),
+			zap.String("repo", fs.bundle.RepoID),
+			zap.String("bundle", fs.bundle.BundleID),
+			zap.Uint64("inode", uint64(r.Inode)),
+		)
+		return
+	}
+	fs.l.Info("Start",
+		zap.String("Request", opType.String()),
+		zap.String("repo", fs.bundle.RepoID),
+		zap.String("bundle", fs.bundle.BundleID),
+		zap.Any("op", op),
+	)
+}
+func (fs *readOnlyFsInternal) opEnd(op interface{}, err error) {
+	opType := reflect.TypeOf(op)
+	switch op.(type) {
+	case *fuseops.ReadFileOp:
+		r := op.(*fuseops.ReadFileOp)
+		fs.l.Info("End",
+			zap.String("Request", opType.String()),
+			zap.String("repo", fs.bundle.RepoID),
+			zap.String("bundle", fs.bundle.BundleID),
+			zap.Uint64("inode", uint64(r.Inode)),
+			zap.Error(err),
+		)
+		return
+	case *fuseops.WriteFileOp:
+		w := op.(*fuseops.WriteFileOp)
+		fs.l.Info("End",
+			zap.String("Request", opType.String()),
+			zap.String("repo", fs.bundle.RepoID),
+			zap.String("bundle", fs.bundle.BundleID),
+			zap.Uint64("inode", uint64(w.Inode)),
+			zap.Error(err),
+		)
+		return
+	case *fuseops.ReadDirOp:
+		r := op.(*fuseops.ReadDirOp)
+		fs.l.Info("End",
+			zap.String("Request", opType.String()),
+			zap.String("repo", fs.bundle.RepoID),
+			zap.String("bundle", fs.bundle.BundleID),
+			zap.Uint64("inode", uint64(r.Inode)),
+			zap.Error(err),
+		)
+		return
+	}
+	fs.l.Info("End",
+		zap.String("Request", reflect.TypeOf(op).String()),
+		zap.String("repo", fs.bundle.RepoID),
+		zap.String("bundle", fs.bundle.BundleID),
+		zap.Any("op", op),
+		zap.Error(err),
+	)
+}
+func typeAssertToFsEntry(p interface{}) *fsEntry {
+	fe, ok := p.(fsEntry)
```

Comment:
using the one-return-value form `fe := p.(fsEntry)` will panic without the need for the conditional on `ok`

Reply:
Thanks